force-app/main/default/classes/AffiliationMappingsService.cls

```diff
@@ -56,20 +57,45 @@ public virtual with sharing class AffiliationMappingsService {
     }
 
     /*****************************************************************************
-    * @description Retrieves a list of models representing all Affiliation Mappings
-    * configured in EDA Settings.
-    * @return A list of AffiliationMappingsModel
-    *****************************************************************************/
-    public virtual List<AffiliationMappingsModel> getAffiliationMappingsModels(){
+     * @description Retrieves a list of models representing all Affiliation Mappings
+     * configured in EDA Settings.
+     * @return A list of AffiliationMappingsModel
+     *****************************************************************************/
+    public virtual List<AffiliationMappingsModel> getAffiliationMappingsModels() {
         return this.locateAffiliationMappingsMapper().getAffiliationMappingsModel();
     }
 
     /*****************************************************************************
-    * @description Retrieves an instance of the AffiliationMappingsMapper.
-    * @return The instance of AffiliationMappingsMapper.
-    ******************************************************************************/
+     * @description Checks user permissions, creates a model and passes to
+     * updateAffiliationMappings.
+     *****************************************************************************/
+    public void updateAffiliationMappings(String mappingName, String accRecordType, String conPrimaryAfflField) {
+        Boolean hasPermission = this.locateUserPermissionService().checkViewSetupForCurrentUser();
+        if (hasPermission == true) {
```

Comment:
Why not make this a guard clause instead? If there is an exception thrown, then simply the check without the variable assignment could work as well.